apps/api/src/campaign/campaign.service.ts

```diff
@@ -529,38 +541,13 @@ export class CampaignService {
         )
         throw new InternalServerErrorException(error)
       }
-    }
-    //donation exists but we need to skip because previous status is from later event than the incoming
-    else {
-      Logger.warn(
-        `Skipping update of donation with paymentIntentId: ${paymentData.paymentIntentId}
-        and status: ${newDonationStatus} because the event comes after existing donation with status: ${donation.status}`,
-      )
+      return
     }
 
-    //For successful donations we will also need to link them to user and add donation wish:
-    if (newDonationStatus === DonationStatus.succeeded) {
-      Logger.debug('metadata?.isAnonymous = ' + metadata?.isAnonymous)
-
-      if (metadata?.isAnonymous != 'true') {
-        await this.prisma.donation.update({
-          where: { id: donation.id },
-          data: {
-            person: {
-              connect: {
-                email: paymentData.billingEmail,
-              },
-            },
-          },
-        })
-      }
-
-      Logger.debug('Saving donation wish ' + metadata?.wish)
-
-      if (metadata?.wish) {
-        await this.createDonationWish(metadata.wish, donation.id, campaign.id)
-      }
-    }
```

Comment:
The removed code was intended to be executed for Charge.Succeeded event so that only then to save the donation wish. Unfortunately moving it above doesn't guarantee that. So, please see my comment above that the actual bug is because this code is skipped by the forgotten 'return' on line 500.